slogtest/slogtest.go

```diff
@@ -1,37 +1,50 @@
+// Package slogtest provides utilities for emitting test logs using clog.
+//
+//	func TestExample(t *testing.T) {
+//		ctx := slogtest.Context(t)
+//		clog.FromContext(ctx).With("foo", "bar").Info("hello world")
+//	}
+//
+// This produces the following test output:
+//
+//	=== RUN   TestExample
+//		slogtest.go:24: level=INFO source=/path/to/example_test.go:13 msg="hello world" foo=bar
+//
+// This package is intended to be used in tests only.
 package slogtest
 
 import (
 	"context"
 	"io"
 	"log/slog"
+	"strings"
+	"testing"
 
 	"github.com/chainguard-dev/clog"
 )
 
-var (
-	_ io.Writer = &logAdapter{}
-)
+var _ io.Writer = &logAdapter{}
 
-type logAdapter struct {
-	l Logger
-}
+type logAdapter struct{ l Logger }
 
 func (l *logAdapter) Write(b []byte) (int, error) {
-	l.l.Log(string(b))
+	l.l.Log(strings.TrimSuffix(string(b), "\n"))
 	return len(b), nil
 }
 
-type Logger interface {
-	Log(args ...any)
-	Logf(format string, args ...any)
-}
+var _ Logger = (*testing.T)(nil)
+var _ Logger = (*testing.B)(nil)
+var _ Logger = (*testing.F)(nil)
+
+type Logger interface{ Log(args ...any) }
 
 // TestLogger gets a logger to use in unit and end to end tests.
 // This logger is configured to log at debug level.
 func TestLogger(t Logger) *clog.Logger {
-	return TestLoggerWithOptions(t, &slog.HandlerOptions{
-		Level: slog.LevelDebug,
```

Comment:
keep this? IIRC this lets debug logs surface in test output (which follows the regular `go test` pattern where they only show if the test fails).